lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -245,23 +250,78 @@ deriveBodyRawRepresentable_init(AbstractFunctionDecl *initDecl, void *) {
                                       nullptr, nullptr);
     litPat->setImplicit();
 
-    auto labelItem = CaseLabelItem(litPat);
+    // Will collect any preliminary guards, plus the final assignment.
+    SmallVector<ASTNode, 2> stmts;
+
+    // If there's an @available attribute specifying when the case was
+    // introduced, generate an early return on unavailability, e.g.
+    // "guard #available(iOS 9, *) else { return nil }"
+    for (auto *attr : elt->getAttrs().getAttributes<AvailableAttr>()) {
```

Comment:
I suggest moving the logic that iterates over the active attributes to Attr.cpp as `DeclAttributes::getIntroduced()` to be analogous to `Decl::Attributes::getDeprecated()`. This will put all the attribute traversal logic in one place.

Reply:
`getUnavailable()` skips over `AvailableVersionComparison::PotentiallyUnavailable` attributes; is that a bug in the function or a necessary difference?

Reply:
I think I told you the wrong thing in my second comment above. I do think you should probably move your code to Attr.cpp as DeclAttributes::getIntroduced() rather than change getUnavailable().